nondex-common/src/main/java/edu/illinois/nondex/shuffling/ControlNondeterminism.java

```diff
@@ -95,6 +111,33 @@ public static Configuration getConfiguration() {
 
     }
 
+    private static void initializeNondex() {
+        // TODO (XLXL) Temporary solution to prevent JVM crash. It's not safe to invoke isBooted()
+        //  or getProperty before saveAndRemoveProperties in System.initPhase1 is done
+        if (System.out == null) {
+            return;
+        }
+
+        if (Utils.checkJDKBefore8()) {
+            if (nondex == null) {
+                nondex = NonDex.getInstance();
+            }
+            return;
+        }
+
+        try {
+            Method isBooted = Class.forName("jdk.internal.misc.VM").getDeclaredMethod("isBooted");
+            if ((Boolean)isBooted.invoke(null) && nondex == null && !isCreatingNonDex) {
+                isCreatingNonDex = true;
```

Comment:
Since you have to use this boolean isCreatingNonDex to set the nondex instance, are there some race conditions to be concerned about here? Are these booleans enough to prevent those? Also, is it necessary to set isCreatingNondex back to false afterwards?